src/app_service/service/chat/service.nim

```diff
@@ -90,6 +91,7 @@ QtObject:
   type Service* = ref object of QObject
     events: EventEmitter
     chats: Table[string, ChatDto] # [chat_id, ChatDto]
+    channelGroups: OrderedTable[string, ChannelGroupDto] # [chatGroup_id, ChannelGroupDto]
```

Comment:
@jrainville since you added `channelGroups` (what is perfectly fine) and since `ChannelGroupDto` contains list of chats belonging to that "channel group", we don't need to maintain `chats` table any more, also in theory it may be the case to have 2 chats with the same id in more than a single channel group (if you have 50 communities with 50 channels each it could be that two channels from different communities may have the same id). 

So my suggestion here is to remove `chats` completely and add a private proc (cause it's a safer way in case of trying to access with an unexisting groupId key) like:
```
proc getChannelGroup(self: Service, groupId: string): ChannelGroupDto =
  if(self.channelGroups.hasKey(groupId)):
    return self.channelGroups[groupId]
``` 
and now on each place where `self.chats` was used we can use `self.getChannelGroup(sectionId).chats`.